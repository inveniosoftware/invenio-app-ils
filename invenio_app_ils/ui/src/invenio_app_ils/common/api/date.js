import { DateTime } from 'luxon';
import _get from 'lodash/get';

export const fromISO = stringDate => {
  return DateTime.fromISO(stringDate);
};

/**
 * Converts datetime JSON string to luxon Datetime object
 * @param date
 * @returns {DateTime}
 */
export const toISO = date => {
  return date.toISO();
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDateTime = date => {
  return date.toFormat('yyyy-MM-dd HH:mm');
};

/**
 *  Serializes date for display in short format
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toShortDate = date => {
  return date.toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for UTC date in short format used to
 *  normalise dates for REST API
 *  @param {DateTime} date luxon DateTime
 *  @re
 */
export const toUTCShortDate = date => {
  if (!(typeof date === 'string' || date instanceof DateTime)) {
    throw Error('Wrong date type provided');
  }
  if (typeof date === 'string') {
    date = fromISO(date);
  }
  return date.toUTC().toFormat('yyyy-MM-dd');
};

/**
 *  Serializes date for display in tables
 */
export function dateFormatter({ col, row }) {
  const dateField = _get(row, col.field);
  return dateField ? toShortDate(fromISO(dateField)) : null;
}
