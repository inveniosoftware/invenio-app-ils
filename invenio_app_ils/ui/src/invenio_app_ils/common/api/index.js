export { document } from './documents/document';
export { http } from './base';
export { item } from './items/item';
export { eitem } from './eitems/eitem';
export { loan } from './loans/loan';
export { patron } from './patrons/patron';
export { location } from './locations/location';
export { internalLocation } from './locations/internalLocation';
export { keyword } from './keywords/keyword';
export { series } from './series/series';
export { related } from './related/related';
