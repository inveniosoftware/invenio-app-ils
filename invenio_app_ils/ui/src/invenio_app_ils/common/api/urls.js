export const ApiURLS = {
  loans: {
    loan: '/circulation/loans/:loanPid',
    list: '/circulation/loans/',
    checkout: '/circulation/loans/checkout',
    request: '/circulation/loans/request',
  },
  items: {},
  documents: {},
  locations: {},
  users: {},
};
