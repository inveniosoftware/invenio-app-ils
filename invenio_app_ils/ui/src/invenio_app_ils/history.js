import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

export const goToHandler = path => e => {
  history.push(path);
};

export const goTo = (path, state = {}) => {
  history.push(path, state);
};

export default history;
