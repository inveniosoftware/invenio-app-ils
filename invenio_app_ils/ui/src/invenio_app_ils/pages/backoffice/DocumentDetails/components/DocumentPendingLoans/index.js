import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchPendingLoans } from './state/actions';
import DocumentPendingLoansComponent from './DocumentPendingLoans';

const mapStateToProps = state => ({
  data: state.documentPendingLoans.data,
  error: state.documentPendingLoans.error,
  isLoading: state.documentPendingLoans.isLoading,
  hasError: state.documentPendingLoans.hasError,
});

const mapDispatchToProps = dispatch => ({
  fetchPendingLoans: documentPid => dispatch(fetchPendingLoans(documentPid)),
});

export const DocumentPendingLoans = compose(
  withRouter,
  connect(
    mapStateToProps,
    mapDispatchToProps
  )
)(DocumentPendingLoansComponent);
