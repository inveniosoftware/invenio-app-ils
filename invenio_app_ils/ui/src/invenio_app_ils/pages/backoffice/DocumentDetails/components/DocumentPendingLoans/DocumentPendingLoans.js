import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { ResultsTable } from '../../../../../common/components';
import { loan as loanApi } from '../../../../../common/api';

import {
  loanSearchQueryUrl,
  viewLoanDetailsUrl,
} from '../../../../../common/urls';
import { formatter } from '../../../../../common/components/ResultsTable/formatters';
import { SeeAllButton } from '../../../components/buttons';

export default class DocumentPendingLoans extends Component {
  constructor(props) {
    super(props);
    this.fetchPendingLoans = props.fetchPendingLoans;
    this.showDetailsUrl = viewLoanDetailsUrl;
    this.seeAllUrl = loanSearchQueryUrl;
  }

  componentDidMount() {
    const { document_pid } = this.props.document;
    this.fetchPendingLoans(document_pid);
  }

  _showDetailsHandler = loan_pid =>
    this.props.history.push(this.showDetailsUrl(loan_pid));

  _seeAllButton = () => {
    const { document_pid } = this.props.document;
    const _click = () =>
      this.props.history.push(
        this.seeAllUrl(
          loanApi
            .query()
            .withDocPid(document_pid)
            .withState('PENDING')
            .qs()
        )
      );
    return <SeeAllButton clickHandler={() => _click()} />;
  };

  prepareData(data) {
    return data.hits.map(row => formatter.loan.toTable(row));
  }

  _render_table(data) {
    const rows = this.prepareData(data);
    rows.totalHits = data.total;
    return (
      <ResultsTable
        rows={rows}
        title={'Pending loans requests'}
        rowActionClickHandler={this._showDetailsHandler}
        seeAllComponent={this._seeAllButton()}
        showMaxRows={this.props.showMaxPendingLoans}
      />
    );
  }

  render() {
    const { data, isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>{this._render_table(data)}</Error>
      </Loader>
    );
  }
}

DocumentPendingLoans.propTypes = {
  document: PropTypes.object.isRequired,
  fetchPendingLoans: PropTypes.func.isRequired,
  data: PropTypes.object.isRequired,
  showMaxPendingLoans: PropTypes.number,
};

DocumentPendingLoans.defaultProps = {
  showMaxPendingLoans: 5,
};
