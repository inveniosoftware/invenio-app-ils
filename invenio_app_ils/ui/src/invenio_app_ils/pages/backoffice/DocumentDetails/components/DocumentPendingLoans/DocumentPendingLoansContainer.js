import React, { Component } from 'react';
import { PropTypes } from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { DocumentPendingLoans } from './DocumentPendingLoans';

export class DocumentPendingLoansContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchPendingLoans = props.fetchPendingLoans;
  }

  componentDidMount() {
    this.fetchPendingLoans(this.props.match.params.documentPid);
  }

  render() {
    const { isLoading, data, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <DocumentPendingLoans
            data={data}
            documentPid={this.props.match.params.documentPid}
          />
        </Error>
      </Loader>
    );
  }
}

DocumentPendingLoansContainer.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  data: PropTypes.object.isRequired,
  hasError: PropTypes.bool.isRequired,
  error: PropTypes.object.isRequired,
  fetchPendingLoans: PropTypes.func.isRequired,
};
