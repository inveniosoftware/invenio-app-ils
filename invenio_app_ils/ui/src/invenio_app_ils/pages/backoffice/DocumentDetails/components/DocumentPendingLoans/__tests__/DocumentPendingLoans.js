import React from 'react';
import { shallow, mount } from 'enzyme';
import { Settings } from 'luxon';
import { fromISO } from '../../../../../../common/api/date';
import { BackOfficeRoutes } from '../../../../../../routes/urls';
import DocumentPendingLoans from '../DocumentPendingLoans';
import history from '../../../../../../history';

jest.mock('../../../../../../common/config');

Settings.defaultZoneName = 'utc';
const stringDate = fromISO('2018-01-01T11:05:00+01:00');

describe('DocumentPendingLoans tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const doc = {
    document_pid: 111,
    metadata: {
      document_pid: 111,
      item_pid: 222,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <DocumentPendingLoans
        document={doc}
        data={{ hits: [], total: 0 }}
        fetchPendingLoans={() => {}}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch pending loans on mount', () => {
    const mockedFetchPendingLoans = jest.fn();
    component = mount(
      <DocumentPendingLoans
        document={doc}
        data={{ hits: [], total: 0 }}
        fetchPendingLoans={mockedFetchPendingLoans}
      />
    );
    expect(mockedFetchPendingLoans).toHaveBeenCalledWith(doc.document_pid);
  });

  it('should render show a message with no pending loans', () => {
    component = mount(
      <DocumentPendingLoans
        document={doc}
        data={{ hits: [], total: 0 }}
        fetchPendingLoans={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const message = component
      .find('Message')
      .filterWhere(element => element.prop('data-test') === 'no-results');
    expect(message).toHaveLength(1);
  });

  it('should render pending loans', () => {
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan1',
          metadata: {
            loan_pid: 'loan1',
            document_pid: 'doc1',
            patron_pid: 'patron_1',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan2',
          metadata: {
            loan_pid: 'loan2',
            document_pid: 'doc1',
            patron_pid: 'patron_2',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
      ],
      total: 2,
    };
    component = mount(
      <DocumentPendingLoans
        document={doc}
        data={data}
        fetchPendingLoans={() => {}}
      />
    );

    expect(component).toMatchSnapshot();
    const rows = component
      .find('TableRow')
      .filterWhere(
        element =>
          element.prop('data-test') === 'loan1' ||
          element.prop('data-test') === 'loan2'
      );
    expect(rows).toHaveLength(2);

    const footer = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(0);
  });

  it('should render the see all button when showing only a few pending loans', () => {
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan1',
          metadata: {
            loan_pid: 'loan1',
            document_pid: 'doc1',
            patron_pid: 'patron_1',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan2',
          metadata: {
            loan_pid: 'loan2',
            document_pid: 'doc1',
            patron_pid: 'patron_2',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
      ],
      total: 2,
    };
    component = mount(
      <DocumentPendingLoans
        document={doc}
        data={data}
        fetchPendingLoans={() => {}}
        showMaxPendingLoans={1}
      />
    );

    expect(component).toMatchSnapshot();
    const footer = component
      .find('TableFooter')
      .filterWhere(element => element.prop('data-test') === 'footer');
    expect(footer).toHaveLength(1);
  });

  it('should go to loan details when clicking on a pending loan', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;
    const data = {
      hits: [
        {
          id: 1,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan1',
          metadata: {
            loan_pid: 'loan1',
            document_pid: 'doc1',
            patron_pid: 'patron_1',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
        {
          id: 2,
          updated: stringDate,
          created: stringDate,
          loan_pid: 'loan2',
          metadata: {
            loan_pid: 'loan2',
            document_pid: 'doc1',
            patron_pid: 'patron_2',
            start_date: stringDate,
            end_date: stringDate,
          },
        },
      ],
      total: 2,
    };

    component = mount(
      <DocumentPendingLoans
        document={doc}
        data={data}
        fetchPendingLoans={() => {}}
        showMaxPendingLoans={1}
      />
    );

    const firstId = data.hits[0].loan_pid;
    const button = component
      .find('TableRow')
      .filterWhere(element => element.prop('data-test') === firstId)
      .find('i');
    button.simulate('click');

    const expectedParam = BackOfficeRoutes.loanDetailsFor(firstId);
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam, {});
  });
});
