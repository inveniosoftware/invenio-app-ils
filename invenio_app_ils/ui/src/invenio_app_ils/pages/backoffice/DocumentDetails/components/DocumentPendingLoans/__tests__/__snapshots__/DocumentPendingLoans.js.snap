// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentPendingLoans tests should load the details component 1`] = `
<Loader>
  <Error>
    <ResultsTable
      fixed={false}
      headerActionClickHandler={null}
      headerActionComponent={null}
      rowActionClickHandler={[Function]}
      rows={Array []}
      seeAllComponent={
        <SeeAllButton
          clickHandler={[Function]}
          disabled={false}
          fluid={false}
        />
      }
      showMaxRows={5}
      singleLine={false}
      subtitle=""
      title="Pending loans requests"
    />
  </Error>
</Loader>
`;

exports[`DocumentPendingLoans tests should render pending loans 1`] = `
<DocumentPendingLoans
  data={
    Object {
      "hits": Array [
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "end_date": "2018-01-01T10:05:00.000Z",
          "loan_pid": "loan1",
          "patron_pid": "patron_1",
          "start_date": "2018-01-01T10:05:00.000Z",
          "updated": "2018-01-01T10:05:00.000Z",
        },
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "end_date": "2018-01-01T10:05:00.000Z",
          "loan_pid": "loan2",
          "patron_pid": "patron_2",
          "start_date": "2018-01-01T10:05:00.000Z",
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 2,
    }
  }
  document={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  fetchPendingLoans={[Function]}
  history={[Function]}
  showMaxPendingLoans={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        rowActionClickHandler={[Function]}
        rows={
          Array [
            Object {
              "ID": "loan1",
              "Patron ID": "patron_1",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
            Object {
              "ID": "loan2",
              "Patron ID": "patron_2",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
          ]
        }
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={false}
        subtitle=""
        title="Pending loans requests"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Pending loans requests"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Pending loans requests
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  withRowAction={true}
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Request created"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Request created
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Patron ID"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Patron ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="State"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                State
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  rowActionClickHandler={[Function]}
                  rows={
                    Array [
                      Object {
                        "ID": "loan1",
                        "Patron ID": "patron_1",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                      Object {
                        "ID": "loan2",
                        "Patron ID": "patron_2",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan1"
                        key="loan1"
                      >
                        <tr
                          className=""
                          data-test="loan1"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-loan1"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-loan1"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan1"
                            key="0-loan1"
                          >
                            <td
                              className=""
                              data-test="ID-loan1"
                            >
                              loan1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan1"
                            key="1-loan1"
                          >
                            <td
                              className=""
                              data-test="Request created-loan1"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan1"
                            key="2-loan1"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan1"
                            >
                              patron_1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan1"
                            key="3-loan1"
                          >
                            <td
                              className=""
                              data-test="State-loan1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan2"
                        key="loan2"
                      >
                        <tr
                          className=""
                          data-test="loan2"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-loan2"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-loan2"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan2"
                            key="0-loan2"
                          >
                            <td
                              className=""
                              data-test="ID-loan2"
                            >
                              loan2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan2"
                            key="1-loan2"
                          >
                            <td
                              className=""
                              data-test="Request created-loan2"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan2"
                            key="2-loan2"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan2"
                            >
                              patron_2
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan2"
                            key="3-loan2"
                          >
                            <td
                              className=""
                              data-test="State-loan2"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  seeAllComponent={
                    <SeeAllButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                    />
                  }
                  showMaxRows={5}
                />
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</DocumentPendingLoans>
`;

exports[`DocumentPendingLoans tests should render show a message with no pending loans 1`] = `
<DocumentPendingLoans
  data={
    Object {
      "hits": Array [],
      "total": 0,
    }
  }
  document={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  fetchPendingLoans={[Function]}
  history={[Function]}
  showMaxPendingLoans={5}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        rowActionClickHandler={[Function]}
        rows={Array []}
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={5}
        singleLine={false}
        subtitle=""
        title="Pending loans requests"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Pending loans requests"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Pending loans requests
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Message
              data-test="no-results"
            >
              <div
                className="ui message"
                data-test="no-results"
              >
                There are no 
              </div>
            </Message>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</DocumentPendingLoans>
`;

exports[`DocumentPendingLoans tests should render the see all button when showing only a few pending loans 1`] = `
<DocumentPendingLoans
  data={
    Object {
      "hits": Array [
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "end_date": "2018-01-01T10:05:00.000Z",
          "loan_pid": "loan1",
          "patron_pid": "patron_1",
          "start_date": "2018-01-01T10:05:00.000Z",
          "updated": "2018-01-01T10:05:00.000Z",
        },
        Object {
          "created": "2018-01-01T10:05:00.000Z",
          "end_date": "2018-01-01T10:05:00.000Z",
          "loan_pid": "loan2",
          "patron_pid": "patron_2",
          "start_date": "2018-01-01T10:05:00.000Z",
          "updated": "2018-01-01T10:05:00.000Z",
        },
      ],
      "total": 2,
    }
  }
  document={
    Object {
      "metadata": Object {
        "document_pid": 111,
        "item_pid": 222,
      },
    }
  }
  fetchPendingLoans={[Function]}
  history={[Function]}
  showMaxPendingLoans={1}
>
  <Loader>
    <Error>
      <ResultsTable
        fixed={false}
        headerActionClickHandler={null}
        headerActionComponent={null}
        rowActionClickHandler={[Function]}
        rows={
          Array [
            Object {
              "ID": "loan1",
              "Patron ID": "patron_1",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
            Object {
              "ID": "loan2",
              "Patron ID": "patron_2",
              "Request created": "2018-01-01 10:05",
              "State": undefined,
            },
          ]
        }
        seeAllComponent={
          <SeeAllButton
            clickHandler={[Function]}
            disabled={false}
            fluid={false}
          />
        }
        showMaxRows={1}
        singleLine={false}
        subtitle=""
        title="Pending loans requests"
      >
        <Segment>
          <div
            className="ui segment"
          >
            <Grid>
              <div
                className="ui grid"
              >
                <GridRow>
                  <div
                    className="row"
                  >
                    <GridColumn
                      verticalAlign="middle"
                      width={13}
                    >
                      <div
                        className="middle aligned thirteen wide column"
                      >
                        <Header
                          as="h3"
                          content="Pending loans requests"
                          subheader=""
                        >
                          <h3
                            className="ui header"
                          >
                            Pending loans requests
                            <HeaderSubheader
                              content=""
                            >
                              <div
                                className="sub header"
                              />
                            </HeaderSubheader>
                          </h3>
                        </Header>
                      </div>
                    </GridColumn>
                    <GridColumn
                      textAlign="right"
                      width={3}
                    >
                      <div
                        className="right aligned three wide column"
                      />
                    </GridColumn>
                  </div>
                </GridRow>
              </div>
            </Grid>
            <Table
              as="table"
              selectable={true}
              striped={true}
            >
              <table
                className="ui selectable striped table"
              >
                <ResultsTableHeader
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  withRowAction={true}
                >
                  <TableHeader
                    as="thead"
                  >
                    <thead
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="header"
                      >
                        <tr
                          className=""
                          data-test="header"
                        >
                          <TableHeaderCell
                            as="th"
                            collapsing={true}
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              collapsing={true}
                              width={2}
                            >
                              <th
                                className="collapsing two wide"
                              />
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="ID"
                            width={2}
                          >
                            <TableCell
                              as="th"
                              className=""
                              width={2}
                            >
                              <th
                                className="two wide"
                              >
                                ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Request created"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Request created
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="Patron ID"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                Patron ID
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                          <TableHeaderCell
                            as="th"
                            key="State"
                          >
                            <TableCell
                              as="th"
                              className=""
                            >
                              <th
                                className=""
                              >
                                State
                              </th>
                            </TableCell>
                          </TableHeaderCell>
                        </tr>
                      </TableRow>
                    </thead>
                  </TableHeader>
                </ResultsTableHeader>
                <ResultsTableBody
                  columns={
                    Array [
                      "ID",
                      "Request created",
                      "Patron ID",
                      "State",
                    ]
                  }
                  rowActionClickHandler={[Function]}
                  rows={
                    Array [
                      Object {
                        "ID": "loan1",
                        "Patron ID": "patron_1",
                        "Request created": "2018-01-01 10:05",
                        "State": undefined,
                      },
                    ]
                  }
                >
                  <TableBody
                    as="tbody"
                  >
                    <tbody
                      className=""
                    >
                      <TableRow
                        as="tr"
                        cellAs="td"
                        data-test="loan1"
                        key="loan1"
                      >
                        <tr
                          className=""
                          data-test="loan1"
                        >
                          <TableCell
                            as="td"
                            textAlign="center"
                          >
                            <td
                              className="center aligned"
                            >
                              <Button
                                as="button"
                                circular={true}
                                compact={true}
                                data-test="btn-view-details-loan1"
                                icon="eye"
                                onClick={[Function]}
                              >
                                <button
                                  className="ui circular compact icon button"
                                  data-test="btn-view-details-loan1"
                                  onClick={[Function]}
                                >
                                  <Icon
                                    as="i"
                                    name="eye"
                                  >
                                    <i
                                      aria-hidden="true"
                                      className="eye icon"
                                    />
                                  </Icon>
                                </button>
                              </Button>
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="ID-loan1"
                            key="0-loan1"
                          >
                            <td
                              className=""
                              data-test="ID-loan1"
                            >
                              loan1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Request created-loan1"
                            key="1-loan1"
                          >
                            <td
                              className=""
                              data-test="Request created-loan1"
                            >
                              2018-01-01 10:05
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="Patron ID-loan1"
                            key="2-loan1"
                          >
                            <td
                              className=""
                              data-test="Patron ID-loan1"
                            >
                              patron_1
                            </td>
                          </TableCell>
                          <TableCell
                            as="td"
                            data-test="State-loan1"
                            key="3-loan1"
                          >
                            <td
                              className=""
                              data-test="State-loan1"
                            >
                              -
                            </td>
                          </TableCell>
                        </tr>
                      </TableRow>
                    </tbody>
                  </TableBody>
                </ResultsTableBody>
                <ResultsTableFooter
                  allRowsNumber={2}
                  columnsNumber={4}
                  seeAllComponent={
                    <SeeAllButton
                      clickHandler={[Function]}
                      disabled={false}
                      fluid={false}
                    />
                  }
                  showMaxRows={1}
                >
                  <TableFooter
                    as="tfoot"
                    data-test="footer"
                    fullWidth={true}
                  >
                    <TableHeader
                      as="tfoot"
                      data-test="footer"
                      fullWidth={true}
                    >
                      <tfoot
                        className="full-width"
                        data-test="footer"
                      >
                        <TableRow
                          as="tr"
                          cellAs="td"
                        >
                          <tr
                            className=""
                          >
                            <TableHeaderCell
                              as="th"
                              colSpan={5}
                              textAlign="right"
                            >
                              <TableCell
                                as="th"
                                className=""
                                colSpan={5}
                                textAlign="right"
                              >
                                <th
                                  className="right aligned"
                                  colSpan={5}
                                >
                                  <span>
                                    Showing first 
                                    1
                                     entries of
                                     
                                    2
                                     
                                  </span>
                                  <span>
                                    <SeeAllButton
                                      clickHandler={[Function]}
                                      disabled={false}
                                      fluid={false}
                                    >
                                      <Button
                                        as="button"
                                        onClick={[Function]}
                                        size="small"
                                      >
                                        <button
                                          className="ui small button"
                                          onClick={[Function]}
                                        >
                                          See all
                                        </button>
                                      </Button>
                                    </SeeAllButton>
                                  </span>
                                </th>
                              </TableCell>
                            </TableHeaderCell>
                          </tr>
                        </TableRow>
                      </tfoot>
                    </TableHeader>
                  </TableFooter>
                </ResultsTableFooter>
              </table>
            </Table>
          </div>
        </Segment>
      </ResultsTable>
    </Error>
  </Loader>
</DocumentPendingLoans>
`;
